service/worker/scheduler/workflow.go

```diff
@@ -470,13 +472,21 @@ func (s *scheduler) getNextTimeV2(cacheBase, after time.Time) getNextTimeResult
 	// too far in the past, we ignore it and fill the cache from after.
 	for try := 1; try <= 3; try++ {
 		// Results covers a contiguous time range so we can search in it
-		for _, result := range s.nextTimeCacheV2.Results {
-			if result.Next.After(after) {
-				return result
+		cache := s.nextTimeCacheV2
+		start := cache.StartTime.AsTime()
+		afterOffset := int64(after.Sub(start))
+		for i, nextOffset := range cache.NextTimes {
```

Comment:
Could this block (3 assignments and the for loop) be moved into a separate function?

Reply:
sure